forest/jasmine/data2mobmat.py

```diff
@@ -1,415 +1,1054 @@
+"""This module contains functions to convert raw GPS data to mobility matrix
+"""
+
 import sys
 import math
-import numpy as np
+from typing import Tuple, Union, Optional, List
 from itertools import groupby
 
-## the radius of the earth
-R = 6.371*10**6
+import numpy as np
+import pandas as pd
 
-def unique(anylist):
-    """
-    Args: anylist, an arbitrary list
-    Return: a list of unique items in anylist
-    """
-    # intilize a null list
-    unique_list = []
-    # traverse for all elements
-    for x in anylist:
-        # check if exists in unique_list or not
-        if x not in unique_list:
-            unique_list.append(x)
-    return unique_list
-
-def cartesian(lat,lon):
-    """
-    Args: latitude and longitude of a location, range[-180, 180],
-          each could be a scalar or a vector (1d np.array), but should be of same length(data type)
-    Return: the corresponding cartesian coordiantes ((0,0,0) as geocenter)
+# the radius of the earth
+R = 6.371 * 10 ** 6
+# a threshold
+TOLERANCE = 1e-6
+
+
+def unique(anylist: list) -> list:
```

Comment:
This function has only one line of code and is used exactly once. Do you think we can get rid of it? Or replace with `numpy.unique()`?

Reply:
Yes we can get rid of it